ui/src/templates/components/TemplateViewOverlay.tsx

```diff
@@ -0,0 +1,87 @@
+import React, {PureComponent} from 'react'
+import {connect} from 'react-redux'
+import {withRouter, WithRouterProps} from 'react-router'
+import _ from 'lodash'
+
+// Components
+import ViewOverlay from 'src/shared/components/ViewOverlay'
+
+// Actions
+import {
+  convertToTemplate as convertToTemplateAction,
+  clearExportTemplate as clearExportTemplateAction,
+} from 'src/templates/actions'
+
+// Types
+import {DocumentCreate} from '@influxdata/influx'
+import {AppState} from 'src/types'
+import {RemoteDataState} from 'src/types'
+
+interface OwnProps {
+  params: {id: string}
+}
+
+interface DispatchProps {
+  convertToTemplate: typeof convertToTemplateAction
+  clearExportTemplate: typeof clearExportTemplateAction
+}
+
+interface StateProps {
+  exportTemplate: DocumentCreate
+  status: RemoteDataState
+}
+
+type Props = OwnProps & StateProps & DispatchProps & WithRouterProps
+
+class TemplateExportOverlay extends PureComponent<Props> {
+  public componentDidMount() {
+    const {
+      params: {id},
+      convertToTemplate,
+    } = this.props
+    convertToTemplate(id)
+  }
+
+  public render() {
+    const {exportTemplate, status} = this.props
+
+    return (
+      <ViewOverlay
+        resource={exportTemplate}
+        overlayHeading={this.overlayTitle}
+        onDismissOverlay={this.onDismiss}
+        status={status}
+      />
+    )
+  }
+
+  private get overlayTitle() {
+    const {exportTemplate} = this.props
+    if (exportTemplate) {
+      return exportTemplate.meta.name
+    }
+    return ''
+  }
+
+  private onDismiss = () => {
+    const {router, clearExportTemplate} = this.props
+
+    router.goBack()
+    clearExportTemplate()
+  }
+}
+
+const mstp = (state: AppState): StateProps => ({
+  exportTemplate: state.templates.exportTemplate.item,
+  status: state.templates.exportTemplate.status,
```

Comment:
could exportTemplate ever be undefined? should a lodash get be used here to ensure you dont get an error?

Reply:
It's being set as null in the state, so won't be ever undefined.